go/cli/mcap/cmd/cat.go

```diff
@@ -109,57 +109,68 @@ func printMessages(
+					}
+					transcoders[channel.SchemaID] = transcoder
 				}
-				files, err := protodesc.FileOptions{}.NewFiles(fileDescriptorSet)
+				msgReader.Reset(message.Data)
+				err = transcoder.Transcode(msg, msgReader)
 				if err != nil {
-					return fmt.Errorf("failed to create file descriptor: %w", err)
+					return fmt.Errorf("failed to transcode %s record on %s: %w", schema.Name, channel.Topic, err)
 				}
-				descriptor, err := files.FindDescriptorByName(protoreflect.FullName(schema.Name))
+			case "protobuf":
+				messageDescriptor, ok := descriptors[channel.SchemaID]
+				if !ok {
+					fileDescriptorSet := &descriptorpb.FileDescriptorSet{}
+					if err := proto.Unmarshal(schema.Data, fileDescriptorSet); err != nil {
+						return fmt.Errorf("failed to build file descriptor set: %w", err)
+					}
+					files, err := protodesc.FileOptions{}.NewFiles(fileDescriptorSet)
+					if err != nil {
+						return fmt.Errorf("failed to create file descriptor: %w", err)
+					}
+					descriptor, err := files.FindDescriptorByName(protoreflect.FullName(schema.Name))
+					if err != nil {
+						return fmt.Errorf("failed to find descriptor: %w", err)
+					}
+					messageDescriptor = descriptor.(protoreflect.MessageDescriptor)
+					descriptors[channel.SchemaID] = messageDescriptor
+				}
+				protoMsg := dynamicpb.NewMessage(messageDescriptor.(protoreflect.MessageDescriptor))
+				if err := proto.Unmarshal(message.Data, protoMsg); err != nil {
+					return fmt.Errorf("failed to parse message: %w", err)
+				}
+				bytes, err := protojson.Marshal(protoMsg)
 				if err != nil {
-					return fmt.Errorf("failed to find descriptor: %w", err)
+					return fmt.Errorf("failed to marshal message: %w", err)
 				}
-				messageDescriptor = descriptor.(protoreflect.MessageDescriptor)
-				descriptors[channel.SchemaID] = messageDescriptor
-			}
-			protoMsg := dynamicpb.NewMessage(messageDescriptor.(protoreflect.MessageDescriptor))
-			if err := proto.Unmarshal(message.Data, protoMsg); err != nil {
-				return fmt.Errorf("failed to parse message: %w", err)
-			}
-			bytes, err := protojson.Marshal(protoMsg)
-			if err != nil {
-				return fmt.Errorf("failed to marshal message: %w", err)
-			}
-			if _, err = msg.Write(bytes); err != nil {
-				return fmt.Errorf("failed to write message bytes: %w", err)
-			}
-		case "jsonschema":
-			if _, err = msg.Write(message.Data); err != nil {
-				return fmt.Errorf("failed to write message bytes: %w", err)
+				if _, err = msg.Write(bytes); err != nil {
+					return fmt.Errorf("failed to write message bytes: %w", err)
+				}
+			case "jsonschema":
+				if _, err = msg.Write(message.Data); err != nil {
+					return fmt.Errorf("failed to write message bytes: %w", err)
+				}
+			default:
+				return fmt.Errorf("JSON output only supported for the following encodings - with schema: ros1msg, protobuf, and JSON; schemaless: JSON. Found: %s", schema.Encoding)
```

Comment:
I don't think we need to mention schemaless channels when the channel already has a schema
```suggestion
				return fmt.Errorf("JSON output only supported for ros1msg, protobuf, and json schemas. Found: %s", schema.Encoding)
```